db/seeds.rb

```diff
@@ -1,79 +1,125 @@
+if Rails.env.production?
+  raise "You're in Production! Seeding data is prohibited!"
+end
+
+DEFAULT_PASSWORD = "password"
+
+def colorize(text, code:)
+  "\e[#{code}m#{text}\e[0m"
+end
+
+if ENV["TRUNCATE_ALL"] == "yes"
+  puts "Truncating all data..."
+  system("bin/rails db:truncate_all")
+end
+
 def create_user!(name)
   User.create!(
     email: "#{name}@example.com",
-    password: "password",
-    password_confirmation: "password",
+    password: DEFAULT_PASSWORD,
+    password_confirmation: DEFAULT_PASSWORD,
     confirmed_at: DateTime.current
   )
 end
 
+def attach(filename, to:)
+  to.attach(io: File.open(Rails.root.join("test/fixtures/files", filename)), filename:)
+end
+
 def create_developer!(attributes)
   avatar_name = attributes.delete(:avatar_name)
   Developer.create!(attributes) do |developer|
-    developer.avatar.attach(io: File.open(Rails.root.join("test/fixtures/files/#{avatar_name}")), filename: avatar_name)
+    attach(avatar_name, to: developer.avatar)
   end
 end
 
-def attach(filename, to:)
-  to.attach(io: File.open(Rails.root.join("test/fixtures/files", filename)), filename:)
-end
+puts "Starting to seed data..."
+ApplicationRecord.transaction requires_new: true do
```

Comment:
If this is in a transaction then the entire seed should probably in one.

Reply:
@joemasilotti 
technically, the entire functionality of the seed is inside the transaction since the code above `ApplicationRecord.transaction requires_new: true do` are methods called inside the transaction block.

Reply:
Ah, right! Thanks for pointing that out.